core/src/main/java/com/scalar/db/api/DistributedTransactionProvider.java

```diff
@@ -39,7 +40,10 @@ public interface DistributedTransactionProvider {
    * Creates an instance of {@link TwoPhaseCommitTransactionManager} for the transaction manager.
    *
    * @param config a database config
-   * @return an instance of {@link TwoPhaseCommitTransactionManager} for the transaction manager
+   * @return an instance of {@link TwoPhaseCommitTransactionManager} for the transaction manager. If
+   *     the transaction manager does not support the two-phase commit interface, returns {@code
+   *     null}.
    */
+  @Nullable
   TwoPhaseCommitTransactionManager createTwoPhaseCommitTransactionManager(DatabaseConfig config);
```

Comment:
Made this method return null if the transaction manager implementation doesn't support the two-phase commit interface.